llama_index/tools/query_engine.py

```diff
@@ -52,10 +52,11 @@ def metadata(self) -> ToolMetadata:
     def call(self, *args: Any, **kwargs: Any) -> ToolOutput:
         if args is not None and len(args) > 0:
             query_str = str(args[0])
-        elif kwargs is not None and len(kwargs) > 0:
-            query_str = str(kwargs)
+        elif kwargs is not None and "input" in kwargs:
+            # NOTE: this assumes our default function schema of `input`
+            query_str = kwargs["input"]
         else:
-            raise ValueError("Cannot call query engine without inputs")
+            raise ValueError("Cannot call query engine without specifying `input` parameter.")
```

Comment:
is it a better fallback to cast the kwargs as a string and still pass it in? Just in case the LLM (ReAct or OpenAI) messes up

Reply:
yeah was thinking about that...basically tradeoff between explicitly failing vs. adding more guardrails

Reply:
adding a `resolve_input_errors` toggle on `QueryEngineTool`

Reply:
I like it, makes sense to me